src/sources/eventstoredb_metrics/mod.rs

```diff
@@ -110,15 +115,17 @@ fn eventstoredb(
 
                             Ok(stats) => {
                                 let metrics = stats.metrics(namespace.clone());
+                                let count = metrics.len();
+                                let byte_size = bytes.len();
 
-                                emit!(&EventStoreDbMetricsReceived {
-                                    events: metrics.len(),
-                                    byte_size: bytes.len(),
-                                });
+                                emit!(&EventsReceived { count, byte_size });
 
                                 let mut metrics = stream::iter(metrics).map(Event::Metric);
                                 if let Err(error) = cx.out.send_all(&mut metrics).await {
-                                    error!(message = "Error sending metric.", %error);
+                                    emit!(&StreamClosedError {
+                                        count,
```

Comment:
If `cx.out.send_all()` errors, is it guaranteed that all `count` events were dropped? If not, looks like it could be tricky to keep `count` updated.

Reply:
Considering we don't really have any information about this, maybe considering that they all got lost it "better": from what I've seen on vector until now, we prefer having duplicates than loosing events.
Now considering it's "just" a metric, I don't really know what scenario to consider. Also, when this happens, Vector will shutdown....